### ISSUE TITLE
[Android Bootcamp] Project 0: Todo - Ready for Review

### ISSUE THREAD

**Issue by maygupta** [maygupta/Todo#1]
My Todo app is complete with the required stories. Will work on the optional stories after creating this issue as mentioned in the instructions.

/cc @codepathreview @codepath

**Comment by maygupta** [comment]
Moved to SQLite for data persistence instead of text file.

**Comment by maygupta** [comment]
Completed optional items with improved UI and using dialog fragments instead of activities. Also added support for due date and priority for a todo.

**Comment by maygupta** [comment]
/cc @codepathreview @codepath
Hi team,
Can you acknowledge if you are able to receive these updates. Thanks

**Comment by nidhi1608** [comment]
We have received your submission and we'll be reviewing it in the next few days. Thanks!

**Comment by maygupta** [comment]
Sorted todos by priority using a custom comparator on Todo class. Updated demo

**Comment by maygupta** [comment]
Focus todo input and auto popup keyboard , when user clicks add button to add a new todo.

Didn't popup keyboard when user selects an already added todo to edit it because user might want to edit some other property like due date/priority so bringing up keyboard for todo input box didn't sound good.

**Comment by maygupta** [comment]
Added icons for action buttons in dialog fragment instead of raw text for better user experience. Updated demo

**Comment by nidhi1608** [comment]
Mayank,

Excellent work! I am glad you got a chance to work on all of the optionals. This was intended in part to give you an introduction to the general rhythm of this course. The course is entirely project based with an app being assigned each week and then due the following week. Each project builds on the last to help each engineer to learn all practical Android development and best practices as quickly as possible. We also do a code review for each submitted project.
Nice job with the project, hope you found it as a helpful introduction.

We are reviewing the submissions right now and will get in touch with you via email a few days after the submission closing date, October 2nd, 2015. In the mean time let us know if you have any questions.

**Comment by maygupta** [comment]
Thanks Nidhi,

I thought of implementing alarm system with a todo where you can set up alarm for a todo from the app itself. If my todo is due for 9 pm today so I can setup the phone alarm directly from the app for a todo. But I wasn't sure if this will complicate the app and for the sake of simplicity of the app, I didn't go in that direction. What are your thoughts on this ? Also I am not sure how much complex that work would be so it will be good to know from you.

**Comment by nidhi1608** [comment]
Thats a great idea if you want to play around with the framework a bit more. Since you have almost three weeks before the class starts, the possibilities of extending your app are endless. However, from the UX standpoint, it might be an overkill. Imagine setting your alarm off for every item if you have a shopping list. You can certainly set the alarm off for the most important tasks. For everything else, you can notify users using simple push notifications. Check out this guide on [working with notifications](http://guides.codepath.com/android/Notifications).

**Comment by maygupta** [comment]
Yes agreed push notifications would be better instead of alarming system. I will look into building it and probably have a gentle reminder/notification to user every day in the morning to remind him of the incomplete todos of the previous day or the other way to do it would be notify the user at say 10pm and remind him that some todos are pending for the day and day is about to end. Do you want to move these to tomorrow.

**Comment by nidhi1608** [comment]
I think both your suggestions would work just fine. If you want to me lenient with the user, you can push them to the next day. :)

**Comment by maygupta** [comment]
Hey @nidhi1608 , just curious to know the plan for the training and is the shortlisting been done ?

**Comment by nidhi1608** [comment]
Mayank, we are in the process of finalizing the list of participants. I'll let you know sometime today or tomorrow. Thanks for following up.

**Comment by maygupta** [comment]
@nidhi1608 
Hi Nidhi, I am unable to access http://courses.codepath.com/sessions

**Comment by nesquena** [comment]
@maygupta can you try again? are you able to access now?

**Comment by maygupta** [comment]
Yes I can access now. Thanks
